filebeat/harvester/reader.go

```diff
@@ -71,6 +74,12 @@ func (r *logFileReader) Read(buf []byte) (int, error) {
 	}
 
 	for {
+		select {
+		case <-r.done:
+			return 0, nil
+		default:
+		}
+
 		n, err := r.fs.Read(buf)
```

Comment:
The file should be closed by the defer function in the harvester. If read is blocking, not sure what we can do. We would need a timeout?